src/Compilers/Core/Portable/Diagnostic/Diagnostic_SimpleDiagnostic.cs

```diff
@@ -130,11 +130,15 @@ public override bool Equals(object obj)
 
             public override int GetHashCode()
             {
-                return Hash.Combine(_descriptor,
-                        Hash.Combine(_messageArgs.GetHashCode(),
-                         Hash.Combine(_location.GetHashCode(),
-                          Hash.Combine(_severity.GetHashCode(), _warningLevel)
-                        )));
+                int hash = 0;
+
+                hash = Hash.Combine(_location.GetHashCode(), 0);
+                hash = Hash.Combine((int)_severity, hash);
```

Comment:
If possible yes, otherwise I can take care of in a separate change. For now, you can assume _location is non-null as all of it's call site currently ensure that.